sql/tests: TestRandomSyntaxSQLSmith failed

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20944917?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20944917?buildTab=artifacts#/) on release-26.1 @ [799b47379d2a135f5608aa57dd5726cee178e8d1](https://github.com/cockroachdb/cockroach/commits/799b47379d2a135f5608aa57dd5726cee178e8d1):


```
    rsg_test.go:804: CREATE TABLE table_5 (col5_0 INET NOT NULL, 😳col5_1 BOX2D NOT NULL, col5_2 NAME NOT NULL, PRIMARY KEY (col5_2 DESC, 😳col5_1 ASC, col5_0 DESC), UNIQUE (col5_0, 😳col5_1, col5_2 ASC), INDEX (col5_2 ASC, col5_0 DESC), UNIQUE (lower(col5_2) DESC), INDEX (col5_0 DESC) NOT VISIBLE, INDEX (col5_2) WHERE table_5.col5_2 >= '':::STRING:::NAME, INDEX (col5_0, 😳col5_1 ASC, lower(CAST(😳col5_1 AS STRING))), INDEX (😳col5_1, col5_0) WHERE table_5.col5_2 >= '':::STRING:::NAME NOT VISIBLE, INDEX (col5_2, col5_0) WHERE table_5.col5_2 > e'\U00002603':::STRING:::NAME);
    rsg_test.go:804: ALTER TABLE table_1 INJECT STATISTICS e'[{"avg_size": 21, "columns": ["\\"coL1_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2, "histo_buckets": [{"distinct_range": 0, "num_eq": 3141170853347280384, "num_range": 0, "upper_bound": "3631701500"}], "histo_col_type": "OID", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 10}, {"avg_size": 8, "columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5, "histo_buckets": [{"distinct_range": 0, "num_eq": 2000000, "num_range": 0, "upper_bound": "1990-03-06 15:25:53.000322+00"}, {"distinct_range": 0, "num_eq": 2411414596668107264, "num_range": 6199106813313898496, "upper_bound": "1989-11-06 19:32:18.000277+00"}, {"distinct_range": 0, "num_eq": 30000000000, "num_range": 50000, "upper_bound": "2020-06-20 09:03:03.000439+00"}, {"distinct_range": 2115567844036728800, "num_eq": 1874477025082680320, "num_range": 4848999345550438400, "upper_bound": "2029-09-30 03:04:51.000087+00"}, {"distinct_range": 15355157.354398295, "num_eq": 0, "num_range": 40000000, "upper_bound": "1991-09-25 02:38:25.000525+00"}, {"distinct_range": 0, "num_eq": 60000, "num_range": 90000, "upper_bound": "1984-06-03 07:29:38.000299+00"}, {"distinct_range": 7580768.527639827, "num_eq": 5000000000, "num_range": 9000000, "upper_bound": "2006-06-13 21:03:51.000893+00"}, {"distinct_range": 6231264871915340000, "num_eq": 3461364020523175424, "num_range": 9190281118916875264, "upper_bound": "2001-01-01 00:00:00+00 BC"}], "histo_col_type": "TIMESTAMPTZ", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 10}, {"avg_size": 14, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5, "histo_col_type": "TIMESTAMP", "histo_version": 3, "name": "__auto__", "null_count": 5, "row_count": 10}, {"avg_size": 6, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 10}, {"avg_size": 23, "columns": ["\\"\\tcol1_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 10}, {"avg_size": 31, "columns": ["\\"co\'l1_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 9, "histo_col_type": "", "name": "__auto__", "null_count": 8, "row_count": 10}, {"avg_size": 23, "columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2, "histo_buckets": [{"distinct_range": 0, "num_eq": 70000, "num_range": 0, "upper_bound": "0"}, {"distinct_range": 2340754768078938600, "num_eq": 20000000, "num_range": 2412071165505007616, "upper_bound": "2422224128"}, {"distinct_range": 0, "num_eq": 6706199280680524800, "num_range": 5829162931374532608, "upper_bound": "2139014789"}, {"distinct_range": 0, "num_eq": 7635524723045603328, "num_range": 4704993579787591680, "upper_bound": "1696128778"}, {"distinct_range": 3100.193873050372, "num_eq": 9000000, "num_range": 7000, "upper_bound": "89339276"}, {"distinct_range": 400000000, "num_eq": 900, "num_range": 400000000, "upper_bound": "369123025"}, {"distinct_range": 33597677.40259613, "num_eq": 861643948313887104, "num_range": 50000000, "upper_bound": "1124871666"}, {"distinct_range": 844996080792748300, "num_eq": 600000, "num_range": 6237021317633080320, "upper_bound": "4230747258"}, {"distinct_range": 0, "num_eq": 8870818583611712512, "num_range": 500000000, "upper_bound": "255293439"}], "histo_col_type": "REGROLE", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 10}]':::JSONB;
    rsg_test.go:804: ALTER TABLE tabl😶e_2 INJECT STATISTICS e'[{"avg_size": 1, "columns": ["col2_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 132, "histo_col_type": "", "name": "__auto__", "null_count": 603, "row_count": 1000}, {"avg_size": 17, "columns": ["\\"col2%_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 801, "histo_col_type": "", "name": "__auto__", "null_count": 262, "row_count": 1000}, {"avg_size": 16, "columns": ["col\u031B2_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 870, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1000}, {"avg_size": 31, "columns": ["col2_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 867, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1000}, {"avg_size": 0, "columns": ["\\"co-l2_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 450, "histo_col_type": "", "name": "__auto__", "null_count": 902, "row_count": 1000}, {"avg_size": 1, "columns": ["col2_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 910, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1000}, {"avg_size": 24, "columns": ["\\"c\\\\\\\\x63ol2_6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 24, "histo_buckets": [{"distinct_range": 0, "num_eq": 2310068146957507584, "num_range": 0, "upper_bound": ""}, {"distinct_range": 2051643191100095700, "num_eq": 4206846998657835008, "num_range": 7259048392578466816, "upper_bound": "5q.qW"}, {"distinct_range": 785282.0022528543, "num_eq": 500000, "num_range": 900000, "upper_bound": "st.IVue1KSc.nl3iKkQu"}, {"distinct_range": 3992673.5495347576, "num_eq": 5139937334381424640, "num_range": 6000000, "upper_bound": "vJkS.ATMD_.gmbYLB3yS.Z.C4r4"}, {"distinct_range": 9108490178270350000, "num_eq": 5714418051179384832, "num_range": 9108490178270350336, "upper_bound": "KJG.4owEk.r0rPZg._.0GlDM.W1G"}, {"distinct_range": 5641715893587287000, "num_eq": 9152461732789117952, "num_range": 5641715893587287040, "upper_bound": "N5.Q-n.JWQz.VBNQMCA.zCSEF_j.dr"}, {"distinct_range": 389.44068238961535, "num_eq": 30, "num_range": 2000, "upper_bound": "xppohmM2.m5kI.0w9.t_VQw.bio.k.qCS1C9._"}, {"distinct_range": 6011990999176851000, "num_eq": 7602090536878214144, "num_range": 6061522278685481984, "upper_bound": "XaPC4QB.G6U.Nq4O.kYcuUNh.b_vxlzXG.ygEYBpr.99qP.jBf8Z6u.890D"}], "histo_col_type": "LTREE", "histo_version": 3, "name": "__auto__", "null_count": 268, "row_count": 1000}, {"avg_size": 28, "columns": ["col2_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 807, "histo_col_type": "", "name": "__auto__", "null_count": 581, "row_count": 1000}, {"avg_size": 22, "columns": ["\\"co%pl2_8\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 447, "histo_col_type": "", "name": "__auto__", "null_count": 434, "row_count": 1000}, {"avg_size": 2, "columns": ["col2_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 691, "histo_col_type": "", "name": "__auto__", "null_count": 736, "row_count": 1000}, {"avg_size": 14, "columns": ["col2_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 596, "histo_col_type": "", "name": "__auto__", "null_count": 44, "row_count": 1000}, {"avg_size": 21, "columns": ["col2_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 127, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1000}]':::JSONB;
    rsg_test.go:804: ALTER TABLE "ta\\U00018440b\vle_3" INJECT STATISTICS e'[{"avg_size": 5, "columns": ["\\" col3_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3170954423614578101, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6977370984772552266}, {"avg_size": 19, "columns": ["col3_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4535178919842416224, "histo_col_type": "", "name": "__auto__", "null_count": 2057883087908641387, "row_count": 6977370984772552266}, {"avg_size": 0, "columns": ["col3_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6380641509652234437, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6977370984772552266}, {"avg_size": 25, "columns": ["\\"col%753_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3892873658699663360, "histo_buckets": [{"distinct_range": 0, "num_eq": 60, "num_range": 0, "upper_bound": "."}], "histo_col_type": "\\"char\\"", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 6977370984772552266}, {"avg_size": 27, "columns": ["col3_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4416299254727293947, "histo_col_type": "", "name": "__auto__", "null_count": 3759472973469754387, "row_count": 6977370984772552266}, {"avg_size": 9, "columns": ["\\"co\\\\\\\\U0004A051l3_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2115133145266723532, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6977370984772552266}, {"avg_size": 12, "columns": ["\\"c ol3_6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6289476986019947352, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 6977370984772552266}, {"avg_size": 10, "columns": ["col3_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2860980164274151936, "histo_buckets": [{"distinct_range": 0, "num_eq": 5091629126212899840, "num_range": 0, "upper_bound": ""}, {"distinct_range": 700, "num_eq": 4725691781396878336, "num_range": 700, "upper_bound": "\\u0014-"}, {"distinct_range": 0, "num_eq": 1816126487904706048, "num_range": 1000000000, "upper_bound": "Px\\u0003"}, {"distinct_range": 0, "num_eq": 6647964276121067520, "num_range": 943169078815643776, "upper_bound": "&@\\u0008R6j"}, {"distinct_range": 0, "num_eq": 800000, "num_range": 6760616310529491968, "upper_bound": "\\u001eKy\\\\6_{"}, {"distinct_range": 9864542761.540209, "num_eq": 200, "num_range": 10000000000, "upper_bound": ";W91w&C\\u0017"}, {"distinct_range": 289260179057149440, "num_eq": 8739712635130791936, "num_range": 471381392505236928, "upper_bound": "bc\\u0004VH\\u000cZ_"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 6977370984772552266}]':::JSONB;
    rsg_test.go:804: ALTER TABLE table_4 INJECT STATISTICS e'[{"avg_size": 9, "columns": ["\\"co l*4_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 27062800825063073, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 926362492155419329}, {"avg_size": 29, "columns": ["col4_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 127547552047391355, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 926362492155419329}, {"avg_size": 3, "columns": ["\\"co}l4_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 329535970972656246, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 926362492155419329}, {"avg_size": 11, "columns": ["co\u030El4_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 15564159125170852, "histo_col_type": "", "name": "__auto__", "null_count": 465706598675120155, "row_count": 926362492155419329}, {"avg_size": 19, "columns": ["co_l4_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 304276529282202862, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 926362492155419329}, {"avg_size": 30, "columns": ["col4_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 710267119669871615, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 926362492155419329}, {"avg_size": 29, "columns": ["col4_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 605231576588429836, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 926362492155419329}, {"avg_size": 2, "columns": ["col4_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 28781377001605700, "histo_buckets": [{"distinct_range": 0, "num_eq": 8000, "num_range": 0, "upper_bound": "\\u0014L]"}, {"distinct_range": 3654608.3639746243, "num_eq": 0, "num_range": 7000000, "upper_bound": "\U00002603"}, {"distinct_range": 35129903902.593544, "num_eq": 40000000000, "num_range": 40000000000, "upper_bound": "XH\\u0006HD"}, {"distinct_range": 0, "num_eq": 4841605604735211520, "num_range": 80000000, "upper_bound": "\'\\u0017J\\u0008\\u0017}"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 926362492155419329}, {"avg_size": 22, "columns": ["col4_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 68800749398340481, "histo_col_type": "", "name": "__auto__", "null_count": 106420005410835478, "row_count": 926362492155419329}]':::JSONB;
    rsg_test.go:804: ALTER TABLE table_5 INJECT STATISTICS e'[{"avg_size": 8, "columns": ["col5_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1908686808278138112, "histo_buckets": [{"distinct_range": 0, "num_eq": 5749178873659229184, "num_range": 0, "upper_bound": "0.0.0.0/0"}, {"distinct_range": 6001571527837530000, "num_eq": 6000, "num_range": 7375003947115119616, "upper_bound": "98.123.139.133/13"}, {"distinct_range": 24091698.23418216, "num_eq": 0, "num_range": 30000000, "upper_bound": "212.174.102.145/24"}, {"distinct_range": 60, "num_eq": 6269015042267186176, "num_range": 60, "upper_bound": "8115:1475:b504:65ea:1ad2:5616:a4cf:5a24/0"}, {"distinct_range": 0, "num_eq": 4505654433751970304, "num_range": 2000000, "upper_bound": "a758:5c3e:299:c8bd:8c92:2bdf:ff58:8153/72"}, {"distinct_range": 0, "num_eq": 40, "num_range": 8074202389377244160, "upper_bound": "8e99:638c:d8d3:6ba2:59c9:4897:3152:9826/91"}], "histo_col_type": "INET", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 2278450993025014596}, {"avg_size": 10, "columns": ["\U0001F633col5_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1700882279814629120, "histo_col_type": "BOX2D", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 2278450993025014596}, {"avg_size": 20, "columns": ["col5_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 338405542078692992, "histo_col_type": "NAME", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 2278450993025014596}]':::JSONB;
    rsg_test.go:804: ALTER TABLE tabl😶e_2 ADD FOREIGN KEY (col2_10) REFERENCES "ta\\U00018440b\vle_3" (" col3_0");
    rsg_test.go:804: CREATE TYPE rand_typ_0 AS (icyei REFCURSOR, evhc DATE, yvlnd PG_LSN, ufsoip BIT(15), inmf TIMETZ);
    rsg_test.go:804: CREATE TYPE rand_typ_1 AS ENUM ('fn', 'bq', 'bjqm', 'l', 'boj');
    rsg_test.go:804: CREATE TYPE rand_typ_2 AS ENUM ('tkndsw');
    rsg_test.go:804: SET sql_safe_updates = false;;
    rsg_test.go:804: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (9061064965960421568);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (6531018868955375196);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (6225609488373177060);
    rsg_test.go:804: ALTER TABLE seed SPLIT AT VALUES (6953818589586561816);
    rsg_test.go:804: ALTER TABLE seed SCATTER;
    rsg_test.go:806: 
    rsg_test.go:807: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith1982599559
--- FAIL: TestRandomSyntaxSQLSmith (302.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #160597 sql/tests: TestRandomSyntaxSQLSmith failed [C-test-failure O-robot T-sql-foundations branch-release-24.3 release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58455

--- a/pkg/sql/tests/rsg_test.go
+++ b/pkg/sql/tests/rsg_test.go
@@ -774,7 +774,7 @@ func TestRandomSyntaxSQLSmith(t *testing.T) {
 		return err
 	}, func(ctx context.Context, db *verifyFormatDB, r *rsg.RSG) error {
 		s := smither.Generate()
-		err := db.exec(t, ctx, s)
+		err := db.execWithTimeout(t, ctx, s, 125*time.Second)
 		if c := (*crasher)(nil); errors.As(err, &c) {
 			if err := db.exec(t, ctx, "USE defaultdb"); err != nil {
 				t.Fatalf("couldn't reconnect to db after crasher: %v", c)

